Update block design guidelines to include info on setup states

alexislloyd opened the pull request:
Proposed guidelines for when to include a setup state in your block design. Would appreciate feedback on content and validation of markdown syntax.

alexislloyd committed:
Update block design guidelines to include info on setup states

Proposed guidelines for when to include a setup state in your block design. Would appreciate feedback on content and validation of markdown syntax.

chrisvanpatten commented on docs/design/block-design.md:
```diff
@@ -19,6 +19,39 @@ Basic block settings won’t always make sense in the context of the placeholder
+Often a block will use the placeholder content to walk users through a setup process. The setup process gathers information from the user that is needed to render the block. A block’s setup state is indicated with a grey background to provide clear differentiation for the user. Not all blocks have setup states — for example, the paragraph block.
+
+
+![An example of a gallery block’s setup state on a grey background](https://make.wordpress.org/design/files/2018/12/gallery-setup.png)
```
Only needs one newline around the image :)

chrisvanpatten replied:
(Applies to all images in the document, fwiw, but I won't note them all as that would be tedious and annoying!)

chrisvanpatten reviewed (commented):
Small review for markdown syntax. Haven't reviewed content yet but I'm sure it's great 👍

jasmussen commented:
Overall I dig this a lot, thanks for doing this!

I found myself initially confused by the term "setup state", but after reading through the document, not only was it clear to me, but I also realized that the term we've used so far for this type of content — "placeholder" — may not have been ideal, as _nothing is shown on the frontend when saving placeholder content_. 

Still, I do think it would be worth adding, perhaps just in the introductory paragraph or as a separate paragraph, a small narrative glossary of terms. Could be as simple as "A setup state, also sometimes referred to as _placeholder_", or something in that vein.

Also, final thing — the images used in the gallery, do we know that we have permission to use those? If not, I can find a few, or you can probably find some on https://www.pexels.com/. Note to be careful with Unsplash, I believe their terms are not as open source as we need them to be.

jasmussen commented:
By the way, this needs a rebase to fix the merge conflict, if you like, I can do that for you tomorrow.

jasmussen commented:
Rebased!

alexislloyd committed:
Edited block-design.md to reflect comments on #12985

Made edits in response to comments from @jasmussen and @chrisvanpatten one #12985:

- Removed extra new lines around images
- Updated wording to resolve any confusion around setup state vs placeholder nomenclature.

jasmussen reviewed (approved):
I dig it 👍 👍

I'll let Chris chime in as well. 

Also I think once the checks pass it _should_ be fine to merge, because it's a doc, but CC: @youknowriad just to be sure. Things have changed on the merge front up until very recently.

chrisvanpatten reviewed (approved):
This looks great to me!
